src/uu/tail/src/parse.rs

```diff
@@ -78,15 +98,30 @@ pub fn parse_obsolete(src: &str) -> Option<Result<impl Iterator<Item = OsString>
                             Some(n) => n,
                             None => return Some(Err(ParseError::Overflow)),
                         };
-                        options.push(OsString::from(format!("{num}")));
+                        options.push(OsString::from(format!("{sign}{num}")));
                     } else {
                         options.push(OsString::from("-n"));
-                        options.push(OsString::from(format!("{num}")));
+                        options.push(OsString::from(format!("{sign}{num}")));
                     }
                     Some(Ok(options.into_iter()))
                 }
                 Err(_) => Some(Err(ParseError::Overflow)),
             }
+        } else if src.len() == 2 {
```

Comment:
I'm not sure these are corner cases per se. Looking at the syntax `-[num][bcl][f]`, I think the structure should be something like this:
- Parse the sign
- Parse optionally a number with suffix, defaulting to 10
- Parse optionally a mode, defaulting to `l`
- Parse optionally the `f` and set `-f` accoringly.

So there shouldn't be a special case for 2 characters, because that's just the case where the parsing of the number defaults to 10.

Reply:
I just wrote basically the same function to test the new WIP argument parser. There are some differences (no sign parsing is required and it doesn't expand into other arguments), but it might still be helpful here: https://github.com/tertsdiepraam/uutils-args/blob/8616b87f5843ebafcb7312b204c673827fafbc09/tests/coreutils/tail.rs#L61-L94 (it hasn't been tested super extensively yet, so let me know if I made a mistake!)

Reply:
Looks good! I will take a deeper look into it tomorrow. There might be an ambiguity with e.g. `tail -c 20`. GNU's variant will handle it as `tail -20c`, but in our case it might result to "invalid file '20'".
And I have to check if `tail -[zqvf]` is handled by clap

Reply:
I updated the parsing based on the things you mentioned: https://github.com/tertsdiepraam/uutils-args/blob/main/tests/coreutils/tail.rs

It's now very compatible I think. It even only accepts it as the first argument. I also don't need to deal with `-z`, `-q` and `-v`. Basically, I think that expanding to clap arguments was a mistake, because it makes the parsing more difficult and too lenient. It's much easier to just parse separately and fall back to clap if it fails.

Reply:
(This doesn't need to happen in this PR though. What you've got looks like a good improvement!)

Reply:
As you have said, I guess clap will handle \-[fvqz]+ fine. I will remove the handling later today, s.t. it will be closer to your implementation.